src/repositories/timesheet.rs

```diff
@@ -0,0 +1,126 @@
+use crate::common::DbResult;
+use chrono::{NaiveDate, NaiveDateTime};
+use sqlx::postgres::PgPool;
+use sqlx::prelude::FromRow;
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[derive(Debug, FromRow)]
+pub struct Timesheet {
+    pub id: Uuid,
+    pub start_date: NaiveDate,
+    pub end_date: NaiveDate,
+    pub total_hours: f32,
+    pub is_editable: bool,
+    pub manager_note: Option<String>,
+    // foreign keys
+    pub user_id: Uuid,
+    pub company_id: Uuid,
+    pub event_id: Uuid,
+    // timestamps
+    pub created_at: NaiveDateTime,
+    pub edited_at: NaiveDateTime,
+    pub deleted_at: Option<NaiveDateTime>,
+}
+
+#[derive(Debug, FromRow)]
+pub struct TimesheetData {
+    pub start_date: NaiveDate,
+    pub end_date: NaiveDate,
+    pub total_hours: f32,
+    pub is_editable: bool,
+    pub manager_note: Option<String>,
+    // foreign keys
+    pub user_id: Uuid,
+    pub company_id: Uuid,
+    pub event_id: Uuid,
+}
+
+#[derive(Clone)]
+pub struct TimesheetRepository {
+    pub pool: Arc<PgPool>,
+}
+
+impl TimesheetRepository {
+    pub fn new(pool: Arc<PgPool>) -> Self {
+        Self { pool }
+    }
+
+    // CRUD
```

Comment:
We also need a call that retrieves a timesheet with its respective workdays.

Similarly, we need a call that resets all workdays to default values.